test/spec/openapi/schema.js

```diff
@@ -108,6 +108,96 @@ test('support 2xx response', async t => {
   t.same(definedPath.responses['3XX'].description, 'Default Response')
 })
 
+test('support multiple content types responses', async t => {
+  const fastify = Fastify()
+  await fastify.register(fastifySwagger, {
+    openapi: true,
+    routePrefix: '/docs',
+    exposeRoute: true
+  })
+
+  const opt = {
+    schema: {
+      response: {
+        200: {
+          description: 'Description and all status-code based properties are working',
+          content: {
+            'application/json': {
+              schema: {
+                name: { type: 'string' },
+                image: { type: 'string' },
+                address: { type: 'string' }
+              }
+            },
+            'application/vnd.v1+json': {
+              schema: {
+                fullName: { type: 'string' },
+                phone: { type: 'string' }
+              }
+            }
+          }
+        },
+        '4xx': {
+          type: 'object',
+          properties: {
+            name: { type: 'string' }
+          }
+        },
+        300: {
+          age: { type: 'number' }
+        }
+      }
+    }
+  }
+  fastify.get('/', opt, () => {})
+
+  await fastify.ready()
+
+  const swaggerObject = fastify.swagger()
+  const api = await Swagger.validate(swaggerObject)
+  const definedPath = api.paths['/'].get
+  t.same(definedPath.responses['200'].description, 'Description and all status-code based properties are working')
+  t.same(definedPath.responses['200'].content, {
```

Comment:
Use strictSame if possible. same is also passing when additional keys and values are in the object.

Reply:
Ahh, thank you for elaborating on this!